### DISCUSSION TITLE
PCM1808 frequency response and not working with internal ADC

### DISCUSSION THREAD

**Discussion by ballistRC** [pschatzmann/arduino-audio-tools#1537] [Q&A] [answered]
I have had i2s-12s working but I noticed a 6db/octave rolloff.  I ran the SineWave gen to prove if it was the DAC PCM5201 or the PCM1808 were the problem was occuring.  The output with the sineWave gen shows very flat response from 100Hz to 10Khz, I didn't bother to test higher.  I then thought I would pump a signal via the internal ADC (esp32) and see if that was also flat.  Unfortunately I've hit a problem with ADC in to I2s out.  I am using the example provided:
#include "AudioTools.h"
uint16_t sample_rate=44100;
AudioInfo info(44100, 2, 16);
AnalogAudioStream in;
I2SStream out;
StreamCopy copier(out, in); // copy in to out

// Arduino Setup
void setup(void) {
  Serial.begin(115200);
  AudioLogger::instance().begin(Serial, AudioLogger::Info);

  // RX automatically uses port 0 with pin GPIO34
  auto cfgRx = in.defaultConfig(RX_MODE);
  cfgRx.copyFrom(info);
  in.begin(cfgRx);

  // TX on I2S_NUM_1
  auto cfgTx = out.defaultConfig(TX_MODE);
  cfgTx.copyFrom(info);
// cfgTx.port_no = 1;
// cfgTx.sample_rate = sample_rate;
// cfgTx.bits_per_sample = 24;
// cfgTx.i2s_format = I2S_STD_FORMAT;
// cfgTx.is_master = true;
// cfgTx.port_no = 0;
// cfgTx.pin_ws = 5;
// cfgTx.pin_bck = 4;
// cfgTx.pin_data = 18;
// cfgTx.pin_data_rx = 19;
// cfgTx.fixed_mclk = sample_rate * 384;
// cfgTx.pin_mck = 3;
// cfgTx.use_apll = true;
  out.begin(cfgTx);
}

// Arduino loop - copy data
void loop() {
  copier.copy();
}

From the Info messages I see this
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13232
load:0x40080400,len:3028
entry 0x400805e4
[     9][D][esp32-hal-cpu.c:244] setCpuFrequencyMhz(): PLL: 480 / 2 = 240 Mhz, APB: 80000000 Hz
[I] AnalogConfigESP32.h : 50 - I2S_MODE_ADC_BUILT_IN
[I] AnalogAudio.h : 64 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AnalogAudioESP32.h : 47 - virtual bool audio_tools::AnalogDriverESP32::begin(audio_tools::AnalogConfigESP32)
[I] AudioTypes.h : 127 -  sample_rate: 44100 / channels: 2 / bits_per_sample: 16
[I] AnalogAudioESP32.h : 51 - auto_center
[I] AnalogAudioESP32.h : 94 - RX_MODE
[I] AudioTypes.h : 127 - out: sample_rate: 44100 / channels: 2 / bits_per_sample: 16
[I] AudioTypes.h : 127 -  sample_rate: 44100 / channels: 2 / bits_per_sample: 16
[I] I2SConfigESP32.h : 80 - rx/tx mode: TX_MODE
[I] I2SConfigESP32.h : 81 - port_no: 0
[I] I2SConfigESP32.h : 82 - is_master: Master
[I] I2SConfigESP32.h : 83 - sample rate: 44100
[I] I2SConfigESP32.h : 84 - bits per sample: 16
[I] I2SConfigESP32.h : 85 - number of channels: 2
[I] I2SConfigESP32.h : 86 - signal_type: Digital
[I] I2SConfigESP32.h : 88 - i2s_format: I2S_STD_FORMAT
[I] I2SConfigESP32.h : 90 - auto_clear: true
[I] I2SConfigESP32.h : 92 - use_apll: true
[I] I2SConfigESP32.h : 97 - buffer_count:6
[I] I2SConfigESP32.h : 98 - buffer_size:512
[I] I2SConfigESP32.h : 103 - pin_bck: 14
[I] I2SConfigESP32.h : 105 - pin_ws: 15
[I] I2SConfigESP32.h : 107 - pin_data: 22
E (128) I2S: register I2S object to platform failed
[E] I2SESP32.h : 214 - begin - i2s_driver_install
E (140) I2S: i2s_write(2129): TX mode is not enabled
[E] I2SESP32.h : 102 - size_t audio_tools::I2SDriverESP32::writeBytes(const void*, size_t)
E (151) I2S: i2s_write(2129): TX mode is not enabled
[E] I2SESP32.h : 102 - size_t audio_tools::I2SDriverESP32::writeBytes(const void*, size_t)
[I] StreamCopy.h : 397 - try write  - 2 (open 1024 bytes) 
E (173) I2S: i2s_write(2129): TX mode is not enabled
[E] I2SESP32.h : 102 - size_t audio_tools::I2SDriverESP32::writeBytes(const void*, size_t)
[I] StreamCopy.h : 397 - try write  - 3 (open 1024 bytes) 

Anyone have any ideas if I have misssed something?  Looks like the i2s driver is failing for some reason.   I'm trying to get a hilber function working which is when I noticed the rolloff in frequerncy, if anyone has some ideas on this, I would love to hear them

BR
tony

**Comment by pschatzmann** [comment, marked as answer]
You messed up with the ports! Analog must be on port 0, so I2S can only be on port 1.
But you left it to the default with is 0, which is already used, so you created a conflict!

**Reply by ballistRC** [reply]
Great thanks, I thought it had to be something silly, now I see it is reasonably flat plus a bit of noise running a tone in via the ADC, so it seems the original problem has something todo with the PCM1808 which seems to be rolling off :
500Hz      395mV rms
1000Hz    243mV
1500Hz    196mV
2000Hz    161mV
2500Hz    137mV
3000Hz    118mV

All reads are RMS.  I didn't bother to check higher.
I have it configured for ESP32 master PCM1808 Slave running with 48 bit clock.  bitclk 48 will only work in slave mode with Fs*384.
Unlike the problem I reported for the ADC-I2S, this also seems to compile fine for the esp32-S3

**Reply by ballistRC** [reply]
I investigated this further and found the input filter on both channels has the wrong value components.  I the boards I received the alias filter was a 10K resistor and a 120nF cap to ground.  This has a pole around 100Hz and caused a loss of sensivity and bad frequency role off.  My solulation was based on the TI eval board and I changed the values to 100R and a 10nF cap to ground.  The board in question come from Aliexpress and look like this. https://www.aliexpress.com/item/1005005686037209.html?spm=a2g0o.order_list.order_list_main.162.94991802KY9yYd

**Reply by ballistRC** [reply]
After fixing they work well and are flat from around 10Hz to 20Khz
